test/functional/ui/messages_spec.lua

```diff
@@ -1715,6 +1717,56 @@ describe('ui/ext_messages', function()
     }
   end)
 
+  it('clears intro screen when new buffer is active', function()
+    api.nvim_set_current_buf(api.nvim_create_buf(true, false))
+    screen:expect {
+      grid = [[
+      ^                                                                                |
+      {1:~                                                                               }|*23
+    ]],
+    }
+  end)
+
+  it('clears intro screen when new buffer is active in floating window', function()
+    local win_opts = { relative = 'editor', height = 1, width = 5, row = 1, col = 5 }
+    api.nvim_open_win(api.nvim_create_buf(false, false), true, win_opts)
+    screen:expect {
+      grid = [[
+                                                                                      |
+      {1:~    }{8:^     }{1:                                                                      }|
+      {1:~                                                                               }|*22
+    ]],
+    }
+  end)
+
+  it('clears intro screen when initial buffer is active in floating window', function()
+    local win_opts = { relative = 'editor', height = 1, width = 5, row = 1, col = 5 }
+    api.nvim_open_win(api.nvim_get_current_buf(), true, win_opts)
+    screen:expect {
+      grid = [[
+                                                                                      |
+      {1:~    }{8:^     }{1:                                                                      }|
+      {1:~                                                                               }|*22
+    ]],
+    }
+  end)
+
+  it('clears intro screen when initial window is converted to be floating', function()
```

Comment:
This one also passes without the added `w_floating` condition. Because `may_show_floating()` is called before the last `nvim_set_current_win()`. This makes it more effective:
```diff
--- a/test/functional/ui/messages_spec.lua
+++ b/test/functional/ui/messages_spec.lua
@@ -1752,11 +1752,13 @@ describe('ui/ext_messages', function()
   end)

   it('clears intro screen when initial window is converted to be floating', function()
-    local init_win_id = api.nvim_get_current_win()
-    command('split')
-    local win_opts = { relative = 'editor', height = 1, width = 5, row = 1, col = 5 }
-    api.nvim_win_set_config(init_win_id, win_opts)
-    api.nvim_set_current_win(init_win_id)
+    exec_lua([[
+      local init_win_id = vim.api.nvim_get_current_win()
+      vim.cmd('split')
+      local win_opts = { relative = 'editor', height = 1, width = 5, row = 1, col = 5 }
+      vim.api.nvim_win_set_config(init_win_id, win_opts)
+      vim.api.nvim_set_current_win(init_win_id)
+    ]])
     screen:expect {
       grid = [[
```

Didn't check the other tests.

Reply:
Ah, indeed, thanks!

I tried to make test failing without the `w_floating` check, but decided that manual check and this kind of test was enough. The `exec_lua()` approach is much better, obviously.